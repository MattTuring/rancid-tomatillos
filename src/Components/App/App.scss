@import '../../variables.scss';

// General
* {
  color: $primary-font-color;
  font-family: $primary-font;
}

.App {
  background-color: #bdbdbdad;
  height: 100vh;
}

button {
  padding: .5vw;
  font-size: 1.3vw;
  background-color: $button-color;
  border: $primary-color;
  &:hover {
    cursor: pointer;
    background-color: $hover-color;
  }
}

// LOGIN
.login {
  background-color: $primary-color;
  display: flex;
  flex-direction: column;
  text-align: left;
  padding: 1%;
  width: 40vw;
  margin-left: auto;
  margin-right: auto;
  margin-top: -230px;
  label {
    margin-top: 1vw;
    font-size: 1.3vw;
  }
  input {
    height: 30px;
    margin-bottom: 1.3vw;
    color: $primary-color;
    font-size: 1.5vw;
    width: 98%;
  }
  button {
    margin: auto;
    margin-top: 1vw;
    width: 98%;
    font-size: 1.5vw;
  }
}

// NAV
nav {
  background-color: $primary-color;
  display: flex;
  align-items: center;
  padding: 1vw;
  height: 50px;
  margin-top: -40px;
  h1 {
    margin-left: 30%;
    font-size: 4vw;
  }
  button {
    margin-left: 150%;
    width: 10vw;
  }
}

.header-background {
  background-color: grey;
  height: 20vh;
  display: flex;
  flex-direction: column;
  justify-content: center;
}

.film-strip {
  height: 200px;
  background-size: contain;
  background-repeat: repeat-x;
}

// movie container
h2 {
  font-size: 1.9vw;
}

.movie-card {
  background-color: black;
  margin: 35px;
}

.movie-container {
  display: grid;
  grid-template-columns: 1fr 1fr 1fr;
  margin-left: auto;
  margin-right: auto;
  width: 90%;
  text-align: center;
  button {
    margin-bottom: 1vw;
    width: 90%;
  }
}

.movie-img {
  background-size: cover;
  height: 50vh;
  background-repeat: no-repeat;
  width: 25vw;
  margin-left: auto;
  margin-right: auto;
  background-position: center;
}

.rating {
  font-weight: bold;
  font-size: 1.3vw;
  margin: 1vw;
}

// SHOW PAGE
.movie-show-page {
  background-color: $primary-color;
  width: 80%;
  margin: auto;
  text-align: center;
  margin-top: -10vw;
  padding: 1vw;
  button {
    float: right;
  }
  p {
    font-size: 1.3vw;
  }
}

.movie-poster {
  height: 30vw;
  width: 100%;
  background-size: contain;
  background-repeat: no-repeat;
}

.movie-info-container {
  display: flex;
}

// MEDIA

@media only screen and (max-width: 1000px)  {
  .movie-container {
    grid-template-columns: 1fr 1fr;
  }

  .login-button {
    margin-left: 45%;
  }

  .movie-img {
    width: 32vw;
  }
}

@media only screen and (max-width: 660px)  {
  .movie-container {
    grid-template-columns: 1fr;
  }

  .login-button {
    margin-left: 25%;
  }

  .movie-img {
    width: 63vw;
  }
}
