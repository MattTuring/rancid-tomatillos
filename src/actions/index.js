export const addMovies = movies => ({
    type: 'ADD_MOVIES',
    movies
});

export const addRatings = ratings => ({
    type: 'ADD_RATING',
    ratings
});

export const addUserState = userInfo => ({
  type: 'ADD_USER_STATE',
  userInfo
});

export const updateLoggedIn = userInfo => ({
  type: 'UPDATE_IS_LOGGEDIN',
  userInfo
});
