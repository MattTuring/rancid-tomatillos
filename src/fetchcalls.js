export const postUser = (url) => {
    return fetch('https://rancid-tomatillos.herokuapp.com/api/v1/login', {
        method: 'POST',
        headers: {
            'Accept': 'application/json',
            'Content-Type': 'application/json'
        },
        body: JSON.stringify({
            "email": 'lucy@turing.io',
            "password": 'password1',
        })
    }) .then(response => {
        if (!response.ok) {
          throw Error('Error fetching ideas');
        }
        return response.json();
      })
}