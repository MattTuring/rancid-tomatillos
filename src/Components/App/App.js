import React from 'react';
import './App.scss';
import {Route} from 'react-router-dom'
import Form from '../Form/Form'
import Header from '../Header/Header'
import { addMovies } from '../../actions';
import { connect } from 'react-redux';

class App extends React.Component {
  constructor() {
    super()
  }

  componentDidMount() {
    fetch('https://rancid-tomatillos.herokuapp.com/api/v1/movies')
    .then(response => response.json())
      .then(data => {
        this.props.addMovies(data)
      })
  }

  render(){
    return (
      <div className="App">
        <Route
          exact path="/login"
          render={() => {
            return (
              <>
                <Header/>
                <Form />
              </>
            )
          }}
        />
        <Route
          exact path="/"
          render={() => {
            return (
              <>
                <Header/>
              </>
            )
          }}
      />
      </div>
    )
  }
}

const mapDispatchToProps = dispatch => ({
  addMovies: movies => dispatch( addMovies(movies) )
})

export default connect(null, mapDispatchToProps)(App)
