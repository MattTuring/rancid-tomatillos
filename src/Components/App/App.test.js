import React from 'react';
import { shallow } from 'enzyme';
import { App, mapStateToProps, mapDispatchToProps } from './App';
import { retrieveMovies } from '../../fetchcalls';
import { addMovies } from '../../actions';

jest.mock('../../fetchcalls.js')

describe('App', () => {
    beforeEach(() => {
        retrieveMovies.mockImplementation(() => {
            return Promise.resolve([{ movie1: 'name' }, { movie2: 'name' }, { movie3: 'name' }])
        });
    });

    it('should match the App snapshot', () => {
        let wrapper = shallow(<App />);
        expect(wrapper).toMatchSnapshot();
    })
    
    it('should invoke retrieveMovies after componentDidMount triggers', () => {
        let wrapper = shallow(<App />);
        expect(retrieveMovies).toHaveBeenCalled();
    });
    
    describe('mapsStateToProps', () => {
        it('should return only movies and loading properties from the store', () => {
            const mockState = {
                movies: [],
                user: {},
                loading: true
            };
            const expected = {
                movies: [],
                loading: true
            };
            const mappedProps = mapStateToProps(mockState);
    
            expect(mappedProps).toEqual(expected);
        });
    });
    
    describe('mapDispatchToProps', () => {
        it('calls dispatch with an addMovies action when addMovies is called',
            () => {
                const mockDispatch = jest.fn();
                const actionToDispatch = addMovies([{ sample: 'movie' }]);
                const mappedProps = mapDispatchToProps(mockDispatch);
    
                mappedProps.addMovies([{ sample: 'movie' }]);
    
                expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch);
        });
    });
})