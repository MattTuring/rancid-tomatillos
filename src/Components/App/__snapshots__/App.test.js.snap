// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match the App snapshot 1`] = `
<div
  className="App"
>
  <Route
    exact={true}
    path="/login"
    render={[Function]}
  />
  <Route
    exact={true}
    path="/users/:id"
    render={[Function]}
  />
  <Route
    exact={true}
    path="/"
    render={[Function]}
  />
  <Route
    exact={true}
    path="/movies/:id"
    render={[Function]}
  />
</div>
`;
